services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_scheduler.py

```diff
@@ -26,6 +38,22 @@ class ObservationItem(BaseModel):
 router = APIRouter()
 
 
+@retry(
+    wait=wait_random_exponential(max=10),
+    stop=stop_after_delay(1 * _MINUTE),
+    retry=retry_if_result(lambda result: result is False),
+    reraise=False,
+    before_sleep=before_sleep_log(_logger, logging.WARNING, exc_info=True),
+)
+def _retry_toggle_observation(
+    dynamic_sidecars_scheduler: DynamicSidecarsScheduler,
+    node_uuid: NodeID,
+    *,
+    is_disabled: bool,
+) -> bool:
+    return dynamic_sidecars_scheduler.toggle_observation(node_uuid, is_disabled)
```

Comment:
Question here: is it a real toggle? I'm a bit confused here with retrying toggling.. will it go on/off/on/off/on? or is that always enabling or disabling? in the second case I would change the name of that function and call it enable or disable or set_enabled or similar...

Reply:
Leaving the name asside. The retry happens, since the disabling of the observation can fail. This is due to the fact that an observation can already be active.
Now how this is used, It is only used to disable, never to enable

Reply:
well then the method name is very weird...

Reply:
made a small rename